src/frontends/paddle/src/op/group_norm.cpp

```diff
@@ -0,0 +1,128 @@
+#include "openvino/frontend/paddle/node_context.hpp"
+#include "openvino/frontend/paddle/visibility.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+
+std::shared_ptr<ov::Node> get_monotonic_range_along_node_rank(const Output<ov::Node>& value) {
+    if (value.get_partial_shape().rank().is_static()) {
+        size_t reshape_rank = value.get_partial_shape().rank().get_length();
+        std::vector<size_t> range_value;
+        for (size_t i = 1; i < reshape_rank; i++)
+            range_value.push_back(i);
+        return default_opset::Constant::create(element::i64, {range_value.size()}, range_value);
+    }
+    const auto value_shape = std::make_shared<default_opset::ShapeOf>(value);
+    return std::make_shared<default_opset::Range>(default_opset::Constant::create(element::i64, {}, {1}),
+                                                  std::make_shared<default_opset::ShapeOf>(value_shape),
+                                                  default_opset::Constant::create(element::i64, {}, {1}),
+                                                  element::i64);
+}
+
+Output<ov::Node> reshape_channel_shaped_node_to_nchw(const Output<ov::Node>& node,
+                                                     const Output<ov::Node>& expected_rank) {
+    const auto one_const = default_opset::Constant::create(element::i64, Shape{1}, {1});
+    const auto two_const = default_opset::Constant::create(element::i64, Shape{1}, {2});
+    const auto tail_shape_rank = std::make_shared<default_opset::Subtract>(expected_rank, two_const);
+    const auto tail_shape = std::make_shared<default_opset::Broadcast>(one_const, tail_shape_rank);
+    const auto C_dim = std::make_shared<default_opset::ShapeOf>(node);
+    const auto new_shape = std::make_shared<default_opset::Concat>(OutputVector{one_const, C_dim, tail_shape}, 0);
+    return std::make_shared<default_opset::Reshape>(node, new_shape, false);
+}
+
+NamedOutputs group_norm(const NodeContext& node) {
+    auto data = node.get_input("X");
+    size_t num_groups = static_cast<size_t>(node.get_attribute<int32_t>("groups"));
+    auto epsilon = node.get_attribute<float>("epsilon", 1e-5);
+    auto data_layout = node.get_attribute<std::string>("data_layout", "NCHW");
+
+    const auto& pshape = data.get_partial_shape();
+    PADDLE_OP_CHECK(node, pshape.rank().is_static());
+    size_t rank_size = pshape.rank().get_length();
+    PADDLE_OP_CHECK(node, rank_size >= 2, "2-D and above tensors supported only");
+
+    if (data_layout == "NHWC") {
+        auto values = std::vector<size_t>{0, rank_size - 1};
+        for (size_t i = 1; i < rank_size - 1; i++) {
+            values.push_back(i);
+        }
+        auto perm1 = default_opset::Constant::create(element::i64, Shape{rank_size}, values);
+        data = std::make_shared<default_opset::Transpose>(data, perm1);
+    }
+    // The process below creates a shape to which we need to reshape the input before normalization.
```

Comment:
I wrote some annotations illustrate part of my code logic. Pls let me know if somewhere is still not claer.